apps/web-giddh/src/app/home/components/total-overdues/total-overdues-chart.component.ts

```diff
@@ -256,14 +227,54 @@ export class TotalOverduesChartComponent implements OnInit, OnDestroy {
             this.toRequest.from = dates[0];
             this.toRequest.to = dates[1];
             this.toRequest.refresh = false;
-            this.store.dispatch(this._homeActions.getTotalOverdues(this.toRequest.from, this.toRequest.to, this.toRequest.refresh));
+            this.getTotalOverdues();
         }
     }
 
     public refreshChart() {
         this.requestInFlight = true;
         this.toRequest.refresh = true;
-        this.store.dispatch(this._homeActions.getTotalOverdues(this.toRequest.from, this.toRequest.to, this.toRequest.refresh));
+        this.getTotalOverdues();
+    }
+
+    public getTotalOverdues() {
+        this.dataFound = false;
+        this.totalRecievable = 0;
+        this.totalPayable = 0;
+        this.PaybaleDurationAmt = 0
+        this.sundryDebtorResponse = [];
+        this.sundryCreditorResponse = [];
+
+        this._dashboardService.GetClosingBalance('sundrydebtors', this.toRequest.from, this.toRequest.to, this.toRequest.refresh).subscribe(response => {
```

Comment:
Consider renaming `GetClosingBalance` (if possible) as it has only 11 occurrences

Do move both calls to `GetClosingBalance` inside a function which can handle the response for both **sundrydebtors** and **sundrycreditors** based on the value passed as other values are same